### PR TITLE
CR-1149619 Add FPT warning message for Versal devices

### PR DESCRIPTION
**Pull request by dbenusov-xilinx**
Signed-off-by: Daniel Benusovich <dbenusov@xilinx.com>
https://jira.xilinx.com/browse/CR-1149619
If certain nodes are missing upon a Versal device coming online the card cannot be flashed as it is not migrated. This will prevent the user from flashing and interacting with the device. There is no message that informs the user of this issue.

#### Problem solved by the commit
Adds a warning message stating that the Versal device is "not migrated".

#### Bug / issue (if any) fixed, which PR introduced the bug, how it was discovered
Users would attempt to program a device and nothing would work

#### How problem was solved, alternative solutions (if any) and why they were rejected
Added an additional field into the user-space VMR sysfs node to allow for checking the versal boot status.

#### Risks (if any) associated the changes in the commit
Warning message may not be specific enough. Perhaps we should refer them to some documentation?

#### What has been tested and how, request additional testing if necessary
Ubuntu 20.04 VCK5000 (Modified driver to return 0)
```
dbenusov@xsjdbenusov50:/proj/rdi/staff/dbenusov/XRT$ xbutil examine -d 17:00
WARNING: Unexpected xocl version (2.14.0) was found. Expected 2.15.0, to match XRT tools.
***********************************************************
*        WARNING          WARNING          WARNING        *
*             Versal Platform is NOT migrated             *
***********************************************************

---------------------------------------------------
[0000:17:00.1] : xilinx_vck5000_gen4x8_qdma_base_2
---------------------------------------------------
Platform
  XSA Name               : xilinx_vck5000_gen4x8_qdma_base_2
  Platform UUID          : 05DCA096-76CB-730B-8D19-EC1192FBAE3F
  FPGA Name              :
  JTAG ID Code           : 0x0
  DDR Size               : 0 Bytes
  DDR Count              : 0
  Revision               : A
  MFG Date               : 0xcc344c
  Mig Calibrated         : false
  P2P Status             : not supported

Mac Addresses            : 00:0A:35:0A:7F:AE
                         : 00:0A:35:0A:7F:AF

Xclbin UUID
  20000000-0000-0000-005D-E180DFAA47A7

Compute Units
  PL Compute Units
    Index   Name                                              Base_Address    Usage   Status

  PS Compute Units
    Index   Name
```

Ubuntu 20.04 VCK5000 (Normal operation)
```
dbenusov@xsjdbenusov50:/proj/rdi/staff/dbenusov/XRT$ xbutil examine -d 17:00
WARNING: Unexpected xocl version (2.14.0) was found. Expected 2.15.0, to match XRT tools.
---------------------------------------------------
[0000:17:00.1] : xilinx_vck5000_gen4x8_qdma_base_2
---------------------------------------------------
Platform
  XSA Name               : xilinx_vck5000_gen4x8_qdma_base_2
  Platform UUID          : 05DCA096-76CB-730B-8D19-EC1192FBAE3F
  FPGA Name              :
  JTAG ID Code           : 0x0
  DDR Size               : 0 Bytes
  DDR Count              : 0
  Revision               : A
  MFG Date               : 0xcc344c
  Mig Calibrated         : false
  P2P Status             : not supported

Mac Addresses            : 00:0A:35:0A:7F:AE
                         : 00:0A:35:0A:7F:AF

Xclbin UUID
  20000000-0000-0000-005D-E180DFAA47A7

Compute Units
  PL Compute Units
    Index   Name                                              Base_Address    Usage   Status

  PS Compute Units
    Index   Name
```
#### Documentation impact (if any)
Maybe add a description about what a non-migrated card is?

### CODE REVIEW

**Review by stsoe — APPROVED** [review]

File: src/runtime_src/core/common/info_vmr.cpp [hunk 1]
```diff
@@ -79,6 +79,9 @@ get_vmr_status(const xrt_core::device* device, vmr_status_type status)
     case vmr_status_type::boot_on_default:
       status_string = "Boot on default";
       break;
+    case vmr_status_type::has_fpt:
+      status_string = "Has fpt";
```
Comment (xdavidz) [inline src/runtime_src/core/common/info_vmr.cpp:83]: would this "has fpt" and "boot on default" case sensitive and trim the while space before and after? this might be a separate PR if you think it is necessary.

**Review by xdavidz — APPROVED** [review]

**Review by chvamshi-xilinx — APPROVED** [review]
